projects/core/src/occ/adapters/product/occ-product.adapter.ts

```diff
@@ -26,6 +26,30 @@ export class OccProductAdapter implements ProductAdapter {
     protected requestsOptimizer: OccRequestsOptimizerService
   ) {}
 
+  loadRealTimeStock(
+    productCode: string,
+    sapCode: string
+  ): Observable<ProductAvailabilities> {
+    const availabilityUrl = this.occEndpoints.buildUrl(
+      'productAvailabilities',
+      {
+        urlParams: {
+          productCode: productCode,
+          sapCode: sapCode,
+        },
+      }
+    );
+
+    return this.http.get(availabilityUrl).pipe(
+      take(1),
+      distinctUntilChanged(),
```

Comment:
[MINOR]
I have problem to understand why do we need take(1) and distrinctUntilChanged()?
AFAIK `http.get()` emits just a single value and then completes the observable.

See Angular docs:
> In general, an observable can return more than one value over time. An observable from HttpClient always emits a single value and then completes, never to emit again.
(https://v17.angular.io/tutorial/tour-of-heroes/toh-pt6)

Reply:
Agreed Removed